tests/utils.go

```diff
@@ -2132,14 +2128,8 @@ func AddEphemeralCdrom(vmi *v1.VirtualMachineInstance, name string, bus string,
 }
 
 func NewRandomFedoraVMIWitGuestAgent() *v1.VirtualMachineInstance {
-	virtClient, err := kubecli.GetKubevirtClient()
-	PanicOnError(err)
-
-	dnsServerIP, err := getClusterDnsServiceIP(virtClient)
-	PanicOnError(err)
-
-	searchDomains := getVMISeachDomains()
-	networkData := GetCloudInitNetworkData(ipv6MasqueradeAddress, ipv6MasqueradeGateway, dnsServerIP, searchDomains)
+	networkData, err := libnet.CreateDefaultCloudInitNetworkData()
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
Do we want to have `expects` within the factory methods ?

Shouldn't `ExpectWithOffset` be used here ?

Reply:
This func is used by several callers and they do not expect it to return an error.
I think we should change the signature in a separate PR/context.

Regarding the use of "Offset", you are right, however we do not know here what is the stack level of the caller (from the test).

Reply:
> This func is used by several callers and they do not expect it to return an error.
> I think we should change the signature in a separate PR/context.

Right, my first thought was that the factory method should be updated to return the created entity + error . Then realized that would probably introduce changes bigger than the PR ... 

+1 on follow up PR for updating the function's signature.

> 
> Regarding the use of "Offset", you are right, however we do not know here what is the stack level of the caller (from the test).

Oh, right.